pkg/client/client.go

```diff
@@ -578,6 +622,30 @@ func (client *DataSetClient) publishBuffer(buf *buffer.Buffer) {
 	client.BufferPerSessionTopic.Pub(buf, buf.Session)
 }
 
+func (client *DataSetClient) purgeBuffer(buf *buffer.Buffer) {
+	if !buf.HasStatus(buffer.Ready) {
+		// buffer is not ready => skip
+		client.Logger.Warn("Buffer cannot be purged", buf.ZapStats()...)
+		return
+	}
+
+	// sent message to terminate go routines
+	client.BufferPerSessionTopic.Pub(Purge{}, buf.Session)
+	client.eventBundlePerKeyTopic.Pub(Purge{}, buf.Session)
+
+	client.buffersAllMutex.Lock()
+	buf.SetStatus(buffer.Purging)
+	delete(client.buffers, buf.Session)
+	client.buffersAllMutex.Unlock()
+
+	client.addEventsMutex.Lock()
+	client.eventBundlePerKeyTopic.Unsub(client.eventBundleSubscriptionChannels[buf.Session], buf.Session)
+	delete(client.eventBundleSubscriptionChannels, buf.Session)
+	client.addEventsMutex.Unlock()
+
+	client.Logger.Debug("purging buffer", buf.ZapStats()...)
```

Comment:
Perhaps it would be a good idea to include more context in the message, e.g. "Purging buffer since it has last received a message X seconds ago (date time string)"

Reply:
This information is part of the `buf.ZapStats()` - `sinceCreatedAtMs`

```
2024-02-16T06:49:37.730+0100    DEBUG   client/client.go:646    purging buffer  {"uuid": "eb3e7070-36b2-4afa-b840-a7cb4b7beae6", "session": "20c81468-9b5d-4aba-ad62-9521e5a983b4-aa4f8120f15357aaedf19d34b0aaa904", "status": "Purging", "logs": 0, "threads": 0, "events": 0, "bufferLength": 42, "bufferRatio": 0.0000080108642578125, "sinceCreatedAtMs": 30534}
```